newmodel.py

```diff
@@ -482,39 +482,44 @@ def clear_input_form():
 
 from ipywidgets import FileUpload
 import os
-upload = FileUpload()
+#upload = FileUpload(multiple=True)
 class observe_file_upload:
     def __init__(self):
         self.metadata = None
-        self.name = None
+        self.name = []
     def __call__(self,change):
         n = change["name"]
         v = change["new"]
         if n == "metadata":
             # Not sure why this sometimes
             # comes in as a list and sometimes
             # not.
-            if type(v) == list:
-                self.metadata = v[0]
-            else:
-                self.metadata = v
-            self.name = self.metadata["name"]
+            for i in range(len(v)):
+                if type(v) == list:
+                    self.metadata = v[i]
+                else:
+                    self.metadata = v
+                self.name.append(self.metadata["name"])
         elif n == "data":
-            d = v[0]
-            assert type(d) == bytes
-            if re.match(r'^.*\.(zip|tgz|tar.gz)',self.name):
-                dname = os.environ["HOME"]+"/Download"
-            else:
-                model = value=input_params.get('title').lower()
-                dname = os.environ["HOME"]+"/agave-model/input_"+model
-            os.makedirs(dname,exist_ok=True)
-            fname = dname + "/" + self.name
-            with open(fname,"wb") as fd:
-                fd.write(d)
-            if os.path.exists(fname):
-                print("upload of '",fname,"' was successful.",sep='')
-            else:
-                print("upload of '",fname,"' failed.",sep='')
+            for j in range(len(v)):
+                d = v[j]
+                assert type(d) == bytes
+                if re.match(r'^.*\.(zip|tgz|tar.gz)',self.name[j]):
+                    dname = os.environ["HOME"]+"/Download"
```

Comment:
I know this is my code, but I'd prefer os.path.join(os.environ["HOME"], "Download")